app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -335,7 +337,8 @@ def import_uploads
 
         params[:upload_ids].each do |upload_id|
           begin
-            sync_task(::Actions::Katello::Repository::ImportUpload, @repository, upload_id)
+            last_item = params[:upload_ids].last == upload_id
+            sync_task(::Actions::Katello::Repository::ImportUpload, @repository, upload_id, {}, :generate_metadata => last_item && generate_metadata)
```

Comment:
Is this guaranteed to result in a boolean? Seems I can pass `publish_repository=false` where false becomes a string and this evaluates to true when `last_item` is true.